okhttp/src/main/kotlin/okhttp3/EventListener.kt

```diff
@@ -422,6 +422,34 @@ abstract class EventListener {
   ) {
   }
 
+  /**
+   * Invoked when a call fails due to cache rules. e.g. onlyIfCached and no cache entry.
+   *
+   * Events will only be received when a Cache is configured for the client.
+   */
+  open fun cacheFailure(call: Call, response: Response) {
+  }
+
+  /**
+   * Invoked when a result is served from the cache. n.b The response should be used
+   * to determine exactly how the cache came to be used, e.g. with a conditional cache response
+   * from the network.
+   *
+   * Events will only be received when a Cache is configured for the client.
+   */
+  open fun cacheMiss(call: Call, response: Response) {
+  }
+
+  /**
+   * Invoked when a response is served from the network. n.b The response should be used
+   * to determine exactly how the cache came to be used, e.g. with a conditional cache response
+   * from the network.
+   *
+   * Events will only be received when a Cache is configured for the client.
```

Comment:
is `response` the application-layer response? Will it have cacheResponse or networkResponse fields defined?